src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -2141,8 +2209,12 @@ public Collection<SystemIndexDescriptor> getSystemIndexDescriptors(Settings sett
             ConfigConstants.SECURITY_CONFIG_INDEX_NAME,
             ConfigConstants.OPENDISTRO_SECURITY_DEFAULT_CONFIG_INDEX
         );
-        final SystemIndexDescriptor systemIndexDescriptor = new SystemIndexDescriptor(indexPattern, "Security index");
-        return Collections.singletonList(systemIndexDescriptor);
+        final SystemIndexDescriptor securityIndexDescriptor = new SystemIndexDescriptor(indexPattern, "Security index");
```

Comment:
Same here, let's add the descriptor if the feature flag is enabled. As long as everything in here is behind the feature flag then we have the right controls.

Reply:
done.